invokeai/backend/model_manager/config.py

```diff
@@ -120,120 +122,163 @@ class ModelRepoVariant(str, Enum):
     FLAX = "flax"
 
 
+class ModelSourceType(str, Enum):
+    """Model source type."""
+
+    Path = "path"
+    Url = "url"
+    HFRepoID = "hf_repo_id"
+    CivitAI = "civitai"
```

Comment:
Do we want to just treat civit as URL?

Reply:
`source_type` is a mostly internal attribute, serving two purposes:
- Disambiguates relative local path sources from HF Repo IDs, which are otherwise indistinguishable 
- Gives a hint on the shape of `source_api_response`

I think it makes sense to leave it as it's own distinct type. Also, CivitAI URLs are different than an arbitrary URL that just points directly to a file.